File: JavaScript/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    const cnt = {};
    array.forEach(v => cnt[v] = (cnt[v] || 0) +1);
    
    const max = Math.max(...Object.values(cnt));
    const modes = Object.keys(cnt).filter(k => cnt[k] === max);
    
    return modes.length === 1 ? Number(modes[0]) : -1;
}

File: Java/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
class Solution {
    public int solution(int[] array) {
        int[] count = new int[1000];  

        for (int num : array) {
            count[num]++;
        }

        int max = 0;        
        int answer = -1;    
        boolean isDuplicate = false;  

        for (int i = 0; i < count.length; i++) {
            if (count[i] > max) {
                max = count[i];
                answer = i;
                isDuplicate = false; 
            } else if (count[i] == max && max != 0) {
                isDuplicate = true;  
            }
        }

        return isDuplicate ? -1 : answer;
    }
}
